api/src/main/java/io/minio/MinioClient.java

```diff
@@ -4096,6 +4398,51 @@ public void putObject(String bucketName, String objectName, String fileName,  Lo
            InvalidKeyException, NoResponseException, XmlPullParserException, ErrorResponseException,
            InternalException, InvalidArgumentException, InsufficientDataException, InvalidResponseException {
 
+    putObject(bucketName, objectName, fileName, size, headerMap, sse, contentType, false);
+  }
+
+
+   /**
+   * Uploads given file as object in given bucket.
+   * <p>
+   * If the object is larger than 5MB, the client will automatically use a multipart session.
+   * </p>
+   * <p>
+   * If the session fails, the user may attempt to re-upload the object by attempting to create
+   * the exact same object again.
+   * </p>
+   * <p>
+   * If the multipart session fails, abort the uploaded parts automatically.
+   * </p>
+   *
+   * @param bucketName  Bucket name.
+   * @param objectName  Object name to create in the bucket.
+   * @param fileName    File name to upload.
+   * @param size        Size of all the data that will be uploaded.
+   * @param headerMap   Custom/additional meta data of the object.
+   * @param sse         encryption metadata.
+   * @param contentType Content type of the stream.
+   * @param objectLockRetention Object lock retention.
+   *
+   * @throws InvalidBucketNameException  upon invalid bucket name is given
+   * @throws NoSuchAlgorithmException
+   *           upon requested algorithm was not found during signature calculation
+   * @throws IOException                 upon connection error
+   * @throws InvalidKeyException
+   *           upon an invalid access key or secret key
+   * @throws NoResponseException         upon no response from server
+   * @throws XmlPullParserException      upon parsing response xml
+   * @throws ErrorResponseException      upon unsuccessful execution
+   * @throws InternalException           upon internal library error
+   * @throws InvalidArgumentException    upon invalid value is passed to a method.
+   * @throws InsufficientDataException   upon getting EOFException while reading given
+   */
+  public void putObject(String bucketName, String objectName, String fileName, Long size, Map<String, String> headerMap,
+                        ServerSideEncryption sse, String contentType, boolean objectLockRetention)
```

Comment:
As we agreed previously, you are going to send separate PR to have `putObject()` with retention and legal hold. Are you still considering to add in this PR?

Reply:
Yes, that i will send as another PR.

Reply:
If so, why are doing this change here?

Reply:
>If so, why are doing this change here?

This change is needed .
 There are two cases to this flow:
 User creates a bucket with lock function enabled. 
**Case 1** : User can put the object on the bucket  and then at later point of time, set the retenetion. This case is handled in this PR.
**Case 2** :  User asks for retention/legalhold at the time of putting object. This will be handled in another PR.

If we dont have the below overloaded putObject then we wont be able to put the object on 
lock enabled buckets and hence not able to test the PR. 
```
  public void putObject(String bucketName, String objectName, String fileName, Long size, Map<String, String> headerMap,
                        ServerSideEncryption sse, String contentType, boolean objectLockRetention) 

``` 

If you think that case 2 should also be present in this PR , then i can come up with that .

Reply:
You cannot add any changes to putObject() if you plan for separate PR.